woocommerce/woocommerce-ios
# [Bundle products in order form] Validate with bundle min/max size rules - storage layer changes

Part of #11159 


## Why


In previous development, I missed two bundle rules for validation - bundle min/max size (the total number of bundle items). These two values are in the product's `bundle_min_size` and `bundle_max_size` fields in the API. In order to persist these values in the product to validate the bundle in the configuration form, schema updates are required to add these two corresponding attributes to the `Product` entity.

<img width="939" alt="Screenshot 2023-11-15 at 4 24 47 PM" src="https://github.com/woocommerce/woocommerce-ios/assets/1945542/587687e9-1c2e-4a24-b059-c39482facef3">

## How

For easier review, you can ignore https://github.com/woocommerce/woocommerce-ios/pull/11161/commits/32e9e02d21a394b2b4fc2eea078fd5dac1da288b which just contains the new model version without any changes. The main change is to add `bundleMinSize` and `bundleMaxSize` attributes to `Product` entity with `nil` as the default value.

## Testing instructions


Just CI is sufficient, but feel free to run the app from another branch in the logged-in state to make sure the migration works without crashing.

---
